projects/controllers/operators.py

```diff
@@ -164,59 +150,26 @@ def delete_operator(uuid, project_id, experiment_id):
     if operator is None:
         raise NotFound("The specified operator does not exist")
 
-    operator_as_dict = operator.as_dict()
-    delete_dependencies(operator_as_dict["uuid"], operator_as_dict["dependencies"])
+    # check if other operators contains the operator being deleted
+    # in dependencies and remove this operator from dependencies
+    operators = db_session.query(Operator) \
+        .filter_by(experiment_id=experiment_id) \
+        .filter(Operator.uuid != uuid)\
+        .all()
+    for op in operators:
+        if uuid in op.dependencies:
+            dependencies = op.dependencies.remove(uuid)
+            if dependencies is None:
+                dependencies = []
```

Comment:
@shiguemi32 se o dependencies for `None` o `if uuid in op.dependencies` vai dar erro.
Esse if é realmente necessário?

Reply:
O op.dependencies nunca vai ser None, pois o valor default dele é []

Esse 'if dependencies is None' é necessário, pois ao remover um item de uma lista que contém apenas este item, o valor retornado é None